src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -108,9 +110,22 @@ public AzureAppConfigurationProvider(IConfigurationClientManager configClientMan
             _options = options ?? throw new ArgumentNullException(nameof(options));
             _optional = optional;
 
-            IEnumerable<KeyValueWatcher> watchers = options.ChangeWatchers.Union(options.MultiKeyWatchers);
+            IEnumerable<KeyValueWatcher> watchers = options.IndividualKvWatchers.Union(options.FeatureFlagWatchers);
 
-            if (watchers.Any())
+            if (options.RegisterAllEnabled)
+            {
+                if (options.FeatureFlagWatchers.Any())
+                {
+                    TimeSpan minFfWatcherRefreshInterval = options.FeatureFlagWatchers.Min(w => w.RefreshInterval);
+
+                    MinRefreshInterval = minFfWatcherRefreshInterval < options.KvCollectionRefreshInterval ? minFfWatcherRefreshInterval : options.KvCollectionRefreshInterval;
+                }
+                else
+                {
+                    MinRefreshInterval = options.KvCollectionRefreshInterval;
+                }
+            }
+            else if (watchers.Any())
             {
                 MinRefreshInterval = watchers.Min(w => w.RefreshInterval);
             }
```

Comment:
What do you think about this?

```
IEnumerable<KeyValueWatcher> watchers = options.IndividualKvWatchers.Union(options.FeatureFlagWatchers);

bool hasWatchers = watchers.Any();
TimeSpan minWatcherRefreshInterval = hasWatchers ? watchers.Min(w => w.RefreshInterval) : TimeSpan.MaxValue;

if (options.RegisterAllEnabled)
{
    MinRefreshInterval = TimeSpan.FromTicks(Math.Min(minWatcherRefreshInterval.Ticks, options.KvCollectionRefreshInterval.Ticks))
}
else
{
    MinRefreshInterval = hasWatchers ? minWatcherRefreshInterval : RefreshConstants.DefaultRefreshInterval;
}
```

Reply:
Yea I think that helps clear it up, thanks!